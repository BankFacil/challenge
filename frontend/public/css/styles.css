/**
 * Global Reset of all HTML Elements
 *
 * Resetting all of our HTML Elements ensures a smoother
 * visual transition between browsers. If you don't believe me,
 * try temporarily commenting out this block of code, then go
 * and look at Mozilla versus Safari, both good browsers with
 * a good implementation of CSS. The thing is, all browser CSS
 * defaults are different and at the end of the day if visual
 * consistency is what we're shooting for, then we need to
 * make sure we're resetting all spacing elements.
 *
 */
html,
body {
  border: 0;
  font-family: "Helvetica-Neue", "Helvetica", Arial, sans-serif;
  line-height: 1.5;
  margin: 0;
  padding: 0;
}
div,
span,
object,
iframe,
img,
table,
caption,
thead,
tbody,
tfoot,
tr,
tr,
td,
article,
aside,
canvas,
details,
figure,
hgroup,
menu,
nav,
footer,
header,
section,
summary,
mark,
audio,
video {
  border: 0;
  margin: 0;
  padding: 0;
}
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
address,
cit,
code,
del,
dfn,
em,
ins,
q,
samp,
small,
strong,
sub,
sup,
b,
i,
hr,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
legend,
label {
  border: 0;
  font-size: 100%;
  vertical-align: baseline;
  margin: 0;
  padding: 0;
}
article,
aside,
canvas,
figure,
figure img,
figcaption,
hgroup,
footer,
header,
nav,
section,
audio,
video {
  display: block;
}
table {
  border-collapse: separate;
  border-spacing: 0;
}
table caption,
table th,
table td {
  text-align: left;
  vertical-align: middle;
}
a img {
  border: 0;
}
:focus {
  outline: 0;
}
html,
body {
  width: 100%;
  height: 100%;
}
body {
  font-family: 'Lato', sans-serif;
}
h1 {
  font-size: 2em;
  font-weight: 100;
}
.messages {
  display: block;
  overflow: hidden;
  float: left;
  max-width: 230px;
  height: 192px;
  margin-right: 10px;
  border: solid 1px #e0e0e0;
  box-shadow: 0 -1px 1px #efefef;
}
.messages ul.message-list {
  display: block;
  overflow: auto;
  width: auto;
  height: 73%;
  padding: 10px;
  list-style-type: none;
  text-indent: 0;
  color: #333;
  background-color: #f0f0f0;
}
.messages ul.message-list li {
  position: relative;
  display: block;
  width: auto;
  max-width: 400px;
  margin: 5px;
  padding: 5px 10px;
  border-radius: 10px;
  background-color: #fff;
  clear: both;
}
.messages ul.message-list li:after {
  font-size: 0;
  display: block;
  visibility: hidden;
  clear: both;
  height: 0;
  content: ' ';
}
.messages ul.message-list li span.sender {
  font-weight: 700;
}
.messages ul.message-list li.sent {
  background-color: #c0dff1;
}
.messages ul.message-list li.sent:before {
  position: absolute;
  top: 10px;
  left: -8px;
  width: 0;
  height: 0;
  content: '';
  border-top: 15px solid transparent;
  border-right: 8px solid #c0dff1;
  border-bottom: 0 solid transparent;
}
.messages ul.message-list li.received {
  float: right;
}
.messages ul.message-list li.received.error {
  background-color: #ff0000;
  color: #ffffff;
}
.messages ul.message-list li.received.error:before {
  border-left-color: #ff0000;
}
.messages ul.message-list li.received:before {
  position: absolute;
  top: 10px;
  right: -8px;
  width: 0;
  height: 0;
  content: '';
  border-top: 0 solid transparent;
  border-bottom: 20px solid transparent;
  border-left: 10px solid #fff;
}
.messages .message-input {
  position: relative;
  z-index: 9;
  height: 100%;
  border-top: solid 1px #e0e0e0;
  box-shadow: 0 -4px 10px #d0d0d0;
}
.messages .message-input input,
.messages .message-input button {
  position: absolute;
  margin: 0;
  padding: 7px;
  border: none;
}
.messages .message-input input {
  width: 100%;
  left: 0;
  color: #a0a0a0;
}
.messages .message-input button {
  right: 0;
  max-width: 60px;
  background: none;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
}
.messages .message-input button:hover {
  background-color: rgba(0, 0, 0, 0.13);
}
.messages .message-input:after {
  font-size: 0;
  display: block;
  visibility: hidden;
  clear: both;
  height: 0;
  content: ' ';
}
