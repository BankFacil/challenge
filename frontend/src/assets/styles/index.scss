:global {
  @import "normalize";
  @import "fonts";
  @import "colors";
  @import "form";
  @import "breakpoints";

  body{
    background: #f2f2f2;
  }

}
