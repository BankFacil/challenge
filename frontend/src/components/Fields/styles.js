import { css } from 'glamor';

export default css({
    display: 'flex',
    justifyContent: 'space-between',
    width: 330,

    ' .fields__group': {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-between',

        ' >div:nth-child(2)': {
            margin: '30px 0',
        },
    },
});
