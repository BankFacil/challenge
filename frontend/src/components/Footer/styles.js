import { css } from 'glamor';

export default css({
    fontSize: '.6rem',
    maxWidth: '600px',
    margin: '0 auto',
    textAlign: 'center',
});
