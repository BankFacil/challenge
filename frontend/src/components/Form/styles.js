import { css } from 'glamor';

export default css({
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'flex-start',
    maxWidth: 700,
    width: '100%',
    margin: '50px 0',
    '@media screen and (max-width: 630px)': {
        flexDirection: 'column',
        alignItems: 'center',

        ' >div': {
            width: '100%',
            maxWidth: 330,
            minWidth: 330,

            '&:first-child': {
                marginBottom: 50,
            },
        },
    },
});
