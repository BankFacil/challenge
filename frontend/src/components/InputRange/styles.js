import styled from "styled-components";

export const Container = styled.div``;
export const Range = styled.div`
  transform: translateY(11px);
  div {
    display: flex;
    justify-content: space-between;
    span {
      font-size: 0.6rem;
      font-weight: 100;
    }
  }
`;
