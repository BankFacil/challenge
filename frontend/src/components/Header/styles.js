import styled from "styled-components";
import { colors } from "../../styles/index";

export const Wrapper = styled.header`
  background-color: ${colors.mountainMeadow};
  padding: 20px 0;
`;

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  margin: 0 auto;
  a {
    text-decoration: none;
    color: ${colors.concrete};
  }
`;

export const Title = styled.h1``;
export const Logo = styled.img`
  height: 1em;
`;
