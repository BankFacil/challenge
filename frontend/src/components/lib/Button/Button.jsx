/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles';

const Button = ({ text, type, onClick }) => (
    <button
        type={type}
        onClick={onClick}
        {...styles}
        data-testid="button"
    >
        {text}
    </button>
);
Button.propTypes = {
    type: PropTypes.string,
    text: PropTypes.string,
    onClick: PropTypes.func,
};

Button.defaultProps = {
    type: 'button',
    text: '',
    onClick: () => ({}),
};

export default Button;
