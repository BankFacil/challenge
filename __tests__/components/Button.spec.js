import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import Button from '../../src/components/Button';
import {ButtonContainer} from '../../src/components/Button/styles';

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = renderer.create(<Button title="Testing" onPress={() => {}} />).toJSON();
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - Secondary Color', () => {
      const component = renderer.create(<Button title="Testing" onPress={() => {}} secondary />).toJSON();
      expect(component).toMatchSnapshot();
    });

    describe('Styles', () => {
      it('should match to snapshot', () => {
        const component = renderer.create(<ButtonContainer />).toJSON();
        expect(component).toMatchSnapshot();
      });
      it('should match to snapshot - with margin', () => {
        const component = renderer.create(<ButtonContainer margin />).toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('Interaction', () => {
    describe('onPressHandler', () => {
      it('should call onPress', () => {
        const onPressMock = jest.fn();
        const component = shallow(<Button title="Testing" onPress={onPressMock} />);

        component.props().onPress();

        expect(onPressMock).toHaveBeenCalled();
        expect(onPressMock).toHaveBeenCalledTimes(1);
      });
    });
  });
});
