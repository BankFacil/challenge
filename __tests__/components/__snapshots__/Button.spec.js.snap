// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering Styles should match to snapshot - with margin 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3099DD",
      "borderRadius": 0,
      "justifyContent": "center",
      "marginBottom": 6,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 6,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
/>
`;

exports[`Button Rendering Styles should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3099DD",
      "borderRadius": 0,
      "justifyContent": "center",
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
/>
`;

exports[`Button Rendering should match to snapshot - Secondary Color 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFC82B",
      "borderRadius": 0,
      "justifyContent": "center",
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "textTransform": "uppercase",
        },
      ]
    }
  >
    Testing
  </Text>
</View>
`;

exports[`Button Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3099DD",
      "borderRadius": 0,
      "justifyContent": "center",
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "textTransform": "uppercase",
        },
      ]
    }
  >
    Testing
  </Text>
</View>
`;
